plugins/ptp_operator/metrics/logparser.go

```diff
@@ -409,6 +409,44 @@ func (p *PTPEventManager) ParseGMLogs(processName, configName, output string, fi
 	}
 }
 
+// ParseTBCLogs ... parse logs for various events
+func (p *PTPEventManager) ParseTBCLogs(processName, configName, output string, fields []string,
+	ptpStats stats.PTPStats) {
+	// T-BC[1743005894]:[ptp4l.0.config] ens7f0  offset  55 T-BC-STATUS s0
+	// 0    1            2               3       4       5  6           7
+	// T-BC 1743005894   ptp4l.0.config  ens7f0  offset  55 T-BC-STATUS s0
+	if strings.Contains(output, bcStatusIdentifier) {
+		if len(fields) < 8 {
+			log.Errorf("T-BC Status is not in right format %s", output)
+			return
+		}
+	} else {
+		return
+	}
+
+	syncState := GetSyncState(fields[7])
+	offset, _ := strconv.ParseInt(fields[5], 10, 64)
+	iface := getAlias(fields[3])
+	ifaceType := types.IFace(iface)
+
+	ptpStats.CheckSource(types.IFace(iface), configName, processName)
+
+	ptpStatsIface := ptpStats[ifaceType]
```

Comment:
This should be ptpStats[master]
There is only two main Key "master" and "clockrealtime" . Only ts2phc uses acttual interface  as key

Reply:
same for this ptpStats.CheckSource(types.IFace(iface), configName, processName) - use the master constant 

ptpStats.CheckSource(masterclock, configName, processName)